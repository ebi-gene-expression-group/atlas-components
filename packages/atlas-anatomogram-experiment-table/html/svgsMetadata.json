[
  {
    "filename": "kidney.duct.svg",
    "species": "kidney",
    "view": "duct",
    "parent": "nephron",
    "ids": [
      "CL_0002201",
      "CL_1001431",
      "CL_0005011"
    ],
    "colours": [
      "#CDCDCD",
      "#525252",
      "#979797"
    ]
  },
  {
    "filename": "kidney.kidney.svg",
    "species": "kidney",
    "view": "kidney",
    "ids": [
      "link_nephron",
      "UBERON_0001225",
      "UBERON_0000362",
      "UBERON_0004200",
      "UBERON_0001224",
      "UBERON_0000056",
      "UBERON_0001285"
    ],
    "colours": [
      "none",
      "#A2A2A2",
      "#CFCFCF",
      "#4F4F4F",
      "#A2A2A2",
      "#3F3F3F",
      "#1A1A1A"
    ]

  },
  {
    "filename": "kidney.glomerulus.svg",
    "species": "kidney",
    "parent": "nephron",
    "view": "glomerulus",
    "ids": [
      "CL_0000653",
      "CL_0002188",
      "UBERON_0005750",
      "CL_1000452",
      "UBERON_0004212",
      "UBERON_0001287",
      "UBERON_0004640",
      "UBERON_0004639",
      "UBERON_0002335",
      "UBERON_0004135",
      "CL_0000650"
    ],
    "colours": [
      "#808080",
      "#3c3c3c",
      "#979797",
      "#979797",
      "#dcdcdc",
      "#c5c5c5",
      "#cccccc",
      "#CCCCCC",
      "#333333",
      "#9c9c9c",
      "#b0b0b0"
    ]
  },
  {
    "filename": "kidney.nephron.svg",
    "species": "kidney",
    "parent": "kidney",
    "view": "nephron",
    "ids": [
      "link_glomerulus",
      "link_duct",
      "UBERON_0001230",
      "UBERON_0000074",
      "UBERON_0004640",
      "UBERON_0004639",
      "UBERON_0001287",
      "UBERON_0001290",
      "UBERON_0001289",
      "UBERON_0001288",
      "UBERON_0001292",
      "UBERON_0005097",
      "UBERON_0001232",
      "UBERON_0001228",
      "UBERON_0009202",
      "UBERON_0009091"
    ],
    "colours": [
      "none",
      "none",
      "#737373",
      "#565656",
      "#2D2D2D",
      "#2D2D2D",
      "#BDBDBD",
      "#404040",
      "#4E4E4E",
      "#A3A3A3",
      "#686868",
      "#373737",
      "#B3B3B3",
      "#595959",
      "#8e8e8e",
      "#393939"

    ]
  },
  {
    "filename": "anolis_carolinensis.svg",
    "species": "anolis_carolinensis",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0000948",
      "UBERON_0002113",
      "UBERON_0014892"
    ]
  },
  {
    "filename": "arabidopsis_thaliana.svg",
    "species": "arabidopsis_thaliana",
    "view": "",
    "ids": [
      "PO_0025034",
      "PO_0009005",
      "PO_0009046",
      "PO_0009001"
    ]
  },
  {
    "filename": "bos_taurus.svg",
    "species": "bos_taurus",
    "view": "",
    "ids": [
      "UBERON_0001013",
      "UBERON_0014892",
      "UBERON_0002114",
      "UBERON_0000955",
      "UBERON_0002113",
      "UBERON_0002048",
      "UBERON_0001155",
      "UBERON_0000948",
      "UBERON_0002107",
      "UBERON_0002016",
      "UBERON_0001898"
    ]
  },
  {
    "filename": "brachypodium_distachyon.flower_parts.svg",
    "species": "brachypodium_distachyon",
    "view": "flower_parts",
    "ids": [
      "PO_0009066",
      "PO_0009030",
      "PO_0009001",
      "PO_0009009",
      "PO_0009089"
    ]
  },
  {
    "filename": "brachypodium_distachyon.whole_plant.svg",
    "species": "brachypodium_distachyon",
    "view": "whole_plant",
    "ids": [
      "PO_0009010",
      "PO_0009049",
      "PO_0025034"
    ]
  },
  {
    "filename": "gallus_gallus.svg",
    "species": "gallus_gallus",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0000948",
      "UBERON_0002113",
      "UBERON_0002107",
      "UBERON_0001155",
      "UBERON_0002106",
      "UBERON_0002048",
      "UBERON_0014892"
    ]
  },
  {
    "filename": "homo_sapiens.brain.svg",
    "species": "homo_sapiens",
    "view": "brain",
    "ids": [
      "UBERON_0002148",
      "UBERON_0001894",
      "UBERON_0001896",
      "UBERON_0002771",
      "UBERON_0002702",
      "UBERON_0002021",
      "UBERON_0001905",
      "UBERON_0001897",
      "UBERON_0001898",
      "UBERON_0002421",
      "UBERON_0003027",
      "UBERON_0001876",
      "UBERON_0001870",
      "UBERON_0001871",
      "UBERON_0001882",
      "UBERON_0002285",
      "UBERON_0001873",
      "UBERON_0001872",
      "UBERON_0002038",
      "UBERON_0001874",
      "UBERON_0001875",
      "UBERON_0002360",
      "UBERON_0000451",
      "UBERON_0000956",
      "UBERON_0002037",
      "UBERON_0002363",
      "UBERON_0001954",
      "UBERON_0002245"
    ]
  },
  {
    "filename": "homo_sapiens.female.svg",
    "species": "homo_sapiens",
    "view": "female",
    "ids": [
      "UBERON_0001981",
      "UBERON_0000178",
      "UBERON_0001637",
      "CL_0000236",
      "CL_0000084",
      "CL_0000623",
      "CL_0000576",
      "CL_0000233",
      "UBERON_0000966",
      "UBERON_0001831",
      "UBERON_0001013",
      "UBERON_0000948",
      "UBERON_0001135",
      "UBERON_0000955",
      "UBERON_0000310",
      "UBERON_0002369",
      "UBERON_0000029",
      "UBERON_0001134",
      "UBERON_0000992",
      "CL_0000738",
      "UBERON_0001044",
      "UBERON_0003889",
      "UBERON_0000995",
      "UBERON_0000002",
      "UBERON_0001021",
      "UBERON_0006618",
      "UBERON_0012249",
      "UBERON_0002421",
      "UBERON_0000977",
      "UBERON_0002185",
      "UBERON_0003126",
      "UBERON_0002048",
      "UBERON_0002372",
      "UBERON_0000970",
      "UBERON_0001876",
      "UBERON_0001736",
      "UBERON_0001264",
      "UBERON_0002107",
      "UBERON_0001155",
      "UBERON_0002371",
      "UBERON_0001255",
      "UBERON_0000945",
      "UBERON_0002114",
      "UBERON_0001043",
      "UBERON_0002110",
      "UBERON_0002106",
      "UBERON_0002108",
      "UBERON_0001987",
      "UBERON_0001295",
      "UBERON_0000996",
      "UBERON_0000947",
      "UBERON_0000007",
      "UBERON_0007650",
      "UBERON_0001153",
      "UBERON_0001154",
      "UBERON_0002116",
      "UBERON_0002079",
      "UBERON_0002084",
      "UBERON_0002146",
      "UBERON_0002135",
      "UBERON_0001103",
      "UBERON_0002481",
      "UBERON_0007844",
      "UBERON_0000341",
      "UBERON_0001052",
      "UBERON_0001706",
      "UBERON_0001728",
      "UBERON_0002037",
      "UBERON_0002245",
      "UBERON_0000451",
      "UBERON_0001870",
      "UBERON_0000004",
      "UBERON_0001871",
      "UBERON_0000956",
      "UBERON_0002113",
      "UBERON_0001225",
      "UBERON_0001621",
      "UBERON_0002134",
      "UBERON_0002046",
      "UBERON_0000014",
      "UBERON_0000167",
      "UBERON_0001723"
    ]
  },
  {
    "filename": "homo_sapiens.male.svg",
    "species": "homo_sapiens",
    "view": "male",
    "ids": [
      "UBERON_0000956",
      "UBERON_0000977",
      "UBERON_0000955",
      "UBERON_0000948",
      "UBERON_0000310",
      "UBERON_0002369",
      "UBERON_0000029",
      "UBERON_0001013",
      "UBERON_0001134",
      "CL_0000738",
      "UBERON_0001871",
      "UBERON_0006618",
      "UBERON_0001621",
      "UBERON_0002421",
      "UBERON_0001000",
      "UBERON_0000998",
      "UBERON_0000473",
      "UBERON_0001301",
      "UBERON_0000970",
      "UBERON_0002372",
      "UBERON_0002048",
      "UBERON_0001876",
      "UBERON_0003126",
      "UBERON_0002185",
      "UBERON_0001021",
      "UBERON_0002037",
      "UBERON_0002245",
      "UBERON_0002113",
      "UBERON_0001225",
      "UBERON_0002046",
      "UBERON_0002371",
      "UBERON_0001870",
      "UBERON_0000451",
      "UBERON_0000007",
      "UBERON_0000947",
      "UBERON_0007650",
      "UBERON_0002084",
      "UBERON_0001153",
      "UBERON_0002116",
      "UBERON_0001052",
      "UBERON_0000004",
      "UBERON_0001723",
      "UBERON_0002079",
      "UBERON_0002146",
      "UBERON_0002135",
      "UBERON_0000989",
      "UBERON_0001728",
      "UBERON_0002240",
      "UBERON_0000341",
      "UBERON_0002134",
      "UBERON_0001103",
      "UBERON_0002107",
      "UBERON_0000945",
      "UBERON_0002106",
      "UBERON_0002114",
      "UBERON_0002110",
      "UBERON_0001264",
      "UBERON_0001155",
      "UBERON_0002108",
      "UBERON_0001154",
      "UBERON_0001135",
      "UBERON_0001255",
      "UBERON_00024818",
      "UBERON_0007844",
      "UBERON_0001043",
      "UBERON_0001044",
      "UBERON_0001831",
      "UBERON_0001736",
      "UBERON_0000014",
      "UBERON_0000178",
      "UBERON_0001981",
      "UBERON_0001637",
      "CL_0000236",
      "CL_0000084",
      "CL_0000623",
      "CL_0000576",
      "CL_0000233",
      "UBERON_0001954",
      "UBERON_0000966",
      "UBERON_0001706",
      "UBERON_0000167",
      "UBERON_0002367"
    ]
  },
  {
    "filename": "hordeum_vulgare.flower_parts.svg",
    "species": "hordeum_vulgare",
    "view": "flower_parts",
    "ids": [
      "PO_0009001",
      "PO_0009030",
      "PO_0009073",
      "PO_0009072",
      "PO_0009009",
      "PO_0009066"
    ]
  },
  {
    "filename": "hordeum_vulgare.whole_plant.svg",
    "species": "hordeum_vulgare",
    "view": "whole_plant",
    "ids": [
      "PO_0020142",
      "PO_0009005",
      "PO_0009006",
      "PO_0009049",
      "PO_0025034"
    ]
  },
  {
    "filename": "macaca_mulatta.svg",
    "species": "macaca_mulatta",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0000451",
      "UBERON_0002037",
      "UBERON_0000948",
      "UBERON_0002113",
      "UBERON_0002107",
      "UBERON_0000473"
    ]
  },
  {
    "filename": "monodelphis_domestica.svg",
    "species": "monodelphis_domestica",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0002037",
      "UBERON_0000948",
      "UBERON_0002113",
      "UBERON_0002107",
      "UBERON_0000473"
    ]
  },
  {
    "filename": "mus_musculus.brain.svg",
    "species": "mus_musculus",
    "view": "brain",
    "ids": [
      "UBERON_0001896",
      "UBERON_0000956",
      "UBERON_0000369",
      "UBERON_0001894",
      "UBERON_0000007",
      "UBERON_0002037",
      "UBERON_0002298",
      "UBERON_0001891",
      "UBERON_0001897",
      "UBERON_0001898",
      "UBERON_0000004",
      "UBERON_0002259",
      "EFO_0000530"
    ]
  },
  {
    "filename": "mus_musculus.female.svg",
    "species": "mus_musculus",
    "view": "female",
    "ids": [
      "UBERON_0000947",
      "UBERON_0001009",
      "UBERON_0001348",
      "UBERON_0001347",
      "UBERON_0000945",
      "UBERON_0002114",
      "UBERON_0001264",
      "UBERON_0002106",
      "UBERON_0002369",
      "UBERON_0002113",
      "UBERON_0001155",
      "UBERON_0002108",
      "UBERON_0001153",
      "UBERON_0002115",
      "UBERON_0002116",
      "UBERON_0001043",
      "UBERON_0002110",
      "UBERON_0000996",
      "UBERON_0000995",
      "UBERON_0001255",
      "UBERON_0001831",
      "UBERON_0001736",
      "UBERON_0001723",
      "UBERON_0001211",
      "UBERON_0000981",
      "UBERON_0002371",
      "UBERON_0007844",
      "UBERON_0001377",
      "UBERON_0014892",
      "UBERON_0002240",
      "UBERON_0001103",
      "UBERON_0002103",
      "UBERON_0001645",
      "UBERON_0000970",
      "UBERON_0001242",
      "UBERON_0000955",
      "UBERON_0000948",
      "UBERON_0002107",
      "UBERON_0001322",
      "UBERON_0001981",
      "UBERON_0000014",
      "UBERON_0001911",
      "UBERON_0003134",
      "UBERON_0000990",
      "UBERON_0000029",
      "UBERON_0001132",
      "UBERON_0002370",
      "UBERON_0002046",
      "UBERON_0002048",
      "UBERON_0000010",
      "UBERON_0003126"
    ]
  },
  {
    "filename": "mus_musculus.male.svg",
    "species": "mus_musculus",
    "view": "male",
    "ids": [
      "UBERON_0000947",
      "UBERON_0001348",
      "UBERON_0001347",
      "UBERON_0000945",
      "UBERON_0002114",
      "UBERON_0001264",
      "UBERON_0002106",
      "UBERON_0002369",
      "UBERON_0002113",
      "UBERON_0001155",
      "UBERON_0002108",
      "UBERON_0001153",
      "UBERON_0002115",
      "UBERON_0002116",
      "UBERON_0001043",
      "UBERON_0002110",
      "UBERON_0001831",
      "UBERON_0001736",
      "UBERON_0000029",
      "UBERON_0000998",
      "UBERON_0000989",
      "UBERON_0000981",
      "UBERON_0002371",
      "UBERON_0007844",
      "UBERON_0001377",
      "UBERON_0002240",
      "UBERON_0002048",
      "UBERON_0001103",
      "UBERON_0003126",
      "UBERON_0002103",
      "UBERON_0001645",
      "UBERON_0001322",
      "UBERON_0001242",
      "UBERON_0002107",
      "UBERON_0000948",
      "UBERON_0000955",
      "UBERON_0014892",
      "UBERON_0001009",
      "UBERON_0001981",
      "UBERON_0000014",
      "UBERON_0001132",
      "UBERON_0001211",
      "UBERON_0002367",
      "UBERON_0001000",
      "UBERON_0001301",
      "UBERON_0000473",
      "UBERON_0001255",
      "UBERON_0002370",
      "UBERON_0000010",
      "UBERON_0000970",
      "UBERON_0001723"
    ]
  },
  {
    "filename": "oryza_sativa.flower_parts.svg",
    "species": "oryza_sativa",
    "view": "flower_parts",
    "ids": [
      "PO_0009010",
      "PO_0009009",
      "PO_0009089",
      "PO_0009030",
      "PO_0009066"
    ]
  },
  {
    "filename": "oryza_sativa.whole_plant.svg",
    "species": "oryza_sativa",
    "view": "whole_plant",
    "ids": [
      "PO_0025034",
      "PO_0009005",
      "PO_0009049",
      "PO_0009006",
      "PO_0005052"
    ]
  },
  {
    "filename": "liver.liver.svg",
    "species": "liver",
    "view": "liver",
    "ids": [
      "UBERON_0001114",
      "UBERON_0001115",
      "UBERON_0002110",
      "UBERON_0001639",
      "UBERON_0001174",
      "UBERON_0004647"
    ],
    "colours": [
      "#808080",
      "#cccccc",
      "#333333",
      "#1a1a1a",
      "#4d4d4d",
      "#808080"
    ]
  },
  {
    "filename": "placenta.placenta.svg",
    "species": "placenta",
    "view": "placenta",
    "ids": [
      "UBERON_0001296",
      "UBERON_0000453",
      "UBERON_0004027",
      "UBERON_0000305",
      "UBERON_0002066",
      "UBERON_0001310",
      "UBERON_0007106"
    ],
    "colours": [
      "#afafaf",
      "#cccccc",
      "#b1b1b1",
      "#464646",
      "#414141",
      "#757575",
      "#b3b3b3"
    ]
  },
  {
    "filename": "placenta.cells.svg",
    "species": "placenta",
    "view": "cells",
    "parent": "placenta",
    "ids": [
      "CL_2000002",
      "EFO_0010710",
      "CL_3000001",
      "UBERON_0000371",
      "UBERON_0000319",
      "UBERON_0000426",
      "EFO_0010712",
      "EFO_0010711",
      "CL_0002601",
      "EFO_0010708",
      "EFOx_7777777"
    ],
    "colours": [
      "#686666",
      "#a4a4a4",
      "#cecece",
      "#cecece",
      "#8d8d8d",
      "#595959",
      "#a4a4a4",
      "#858585",
      "#d7d7d7",
      "#535353",
      "#cccccc"

    ]
  },
  {
    "filename": "liver.lobule.svg",
    "species": "liver",
    "view": "lobule",
    "parent": "liver",
    "ids": [
      "CL_0000182",
      "CL_0000632",
      "UBERON_0001282",
      "CL_1000488",
      "UBERON_0001193",
      "UBERON_0006841",
      "UBERON_0001639",
      "EFO_0010704",
      "UBERON_0001281",
      "CL_1000398",
      "EFO_0010705",
      "EFO_0010706",
      "CL_0000091"
    ],
    "colours": [
      "#b3b3b3",
      "#333333",
      "#C7c7c7",
      "#4d4d4d",
      "#808080",
      "#C7c7c7",
      "#999999",
      "#333333",
      "#cccccc",
      "#808080",
      "#666666",
      "#4d4d4d",
      "#1a1a1a"
    ]
  },
  {
    "filename": "lung.lung.svg",
    "species": "lung",
    "view": "lung",
    "ids": [
      "UBERON_0002170",
      "UBERON_0002174",
      "UBERON_0002171",
      "UBERON_0008952",
      "UBERON_0008953",
      "UBERON_0003126",
      "UBERON_0002182",
      "UBERON_0002183",
      "UBERON_0002184",
      "UBERON_0002186",
      "UBERON_0002169"
    ],
    "colours": [
      "#c0c0c0",
      "#888888",
      "#505050",
      "#c0c0c0",
      "#494949",
      "#6e6e6e",
      "#3e3e3e",
      "#4D4D4D",
      "#8f8f8f",
      "#3e3e3e",
      "#2b2b2b"
    ]
  },
  {
    "filename": "lung.alveoli_fin.svg",
    "species": "lung",
    "view": "alveoli_fin",
    "parent": "lung",
    "ids": [
      "UBERON_0002299",
      "UBERON_0002188",
      "UBERON_0016405"
    ],
    "colours": [
      "#949494",
      "#bababa",
      "#363636"
    ]
  },
  {
    "filename": "lung.alveoli_section.svg",
    "species": "lung",
    "view": "alveoli_section",
    "parent": "alveoli_fin",
    "ids": [
      "CL_0002062",
      "CL_0002063",
      "UBERON_0003456",
      "EFO_0010667",
      "EFO_0010668",
      "EFO_0010669"
    ],
    "colours": [
      "#7E7E7E",
      "#4F4F4F",
      "#BDBDBD",
      "#3D3D3D",
      "#cccccc",
      "#202020"
    ]
  },
  {
    "filename": "lung.airway.svg",
    "species": "lung",
    "view": "airway",
    "parent": "lung",
    "ids": [
      "CL_0002633",
      "CL_1001567",
      "CL_1000271",
      "CL_0000158",
      "EFO_0010666",
      "CL_0002370",
      "UBERON_0003504",
      "CL_0002598"
    ],
    "colours": [
      "#252525",
      "#666666",
      "#888888",
      "#3D3D3D",
      "#B9B9B9",
      "#B7B7B7",
      "#999999",
      "#464646"
    ]
  },
  {
    "filename": "pancreas.acinus.svg",
    "species": "pancreas",
    "view": "acinus",
    "parent": "pancreas",
    "ids": [
      "CL_0000173",
      "CL_0000171",
      "CL_0000169",
      "CL_0002275",
      "CL_0005019",
      "CL_0002410",
      "CL_0000622"
    ],
    "colours": [
      "#333333",
      "#595959",
      "#cccccc",
      "#222222",
      "#898989",
      "#ababab",
      "#797979"
    ]
  },
  {
    "filename": "pancreas.pancreas.svg",
    "species": "pancreas",
    "view": "pancreas",
    "ids": [
      "UBERON_0001069",
      "UBERON_0001151",
      "UBERON_0001150",
      "UBERON_0001263",
      "UBERON_0000006",
      "UBERON_0004913",
      "UBERON_0001174",
      "UBERON_0007329",
      "link_acinus"
    ],
    "colours": [
      "#8b8b8b",
      "#8b8b8b",
      "#c9c9c9",
      "#3d3d3d",
      "#888888",
      "#2b2b2b",
      "#5c5c5c",
      "#363636",
      "#668000"
    ]
  },
  {
    "filename": "papio_anubis.svg",
    "species": "papio_anubis",
    "view": "",
    "ids": [
      "UBERON_0000948",
      "UBERON_0002107",
      "UBERON_0000945",
      "UBERON_0001155",
      "UBERON_0002371",
      "UBERON_0002113",
      "UBERON_0002106",
      "UBERON_0014892",
      "UBERON_0002370",
      "UBERON_0000955",
      "UBERON_0001871",
      "UBERON_0001870",
      "UBERON_0000007",
      "UBERON_0002048",
      "UBERON_0000029",
      "UBERON_0002037"
    ]
  },
  {
    "filename": "rattus_norvegicus.svg",
    "species": "rattus_norvegicus",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0002113",
      "CL_0000336",
      "UBERON_0001388",
      "UBERON_0000948",
      "UBERON_0002107",
      "UBERON_0002048",
      "UBERON_0002106",
      "UBERON_0002370",
      "UBERON_0000995",
      "UBERON_0000473",
      "UBERON_0001155"
    ]
  },
  {
    "filename": "solanum_lycopersicum.flower_parts.svg",
    "species": "solanum_lycopersicum",
    "view": "flower_parts",
    "ids": [
      "PO_0009084",
      "PO_0009030",
      "PO_0009072",
      "PO_0009073",
      "PO_0009074",
      "PO_0009066",
      "PO_0009010"
    ]
  },
  {
    "filename": "solanum_lycopersicum.whole_plant.svg",
    "species": "solanum_lycopersicum",
    "view": "whole_plant",
    "ids": [
      "PO_0025034",
      "PO_0000056",
      "PO_0009046",
      "PO_0009001",
      "PO_0009005"
    ]
  },
  {
    "filename": "solanum_tuberosum.svg",
    "species": "solanum_tuberosum",
    "view": "",
    "ids": []
  },
  {
    "filename": "sorghum_bicolor.flower_parts.svg",
    "species": "sorghum_bicolor",
    "view": "flower_parts",
    "ids": [
      "PO_0009049",
      "PO_0009051",
      "PO_0009001",
      "PO_0009009",
      "PO_0009089",
      "PO_0009066",
      "PO_0009030"
    ]
  },
  {
    "filename": "sorghum_bicolor.whole_plant.svg",
    "species": "sorghum_bicolor",
    "view": "whole_plant",
    "ids": [
      "PO_0009006",
      "PO_0006079",
      "PO_0009047",
      "PO_0000230",
      "PO_0009049",
      "PO_0025034",
      "PO_0009005"
    ]
  },
  {
    "filename": "tetraodon_nigroviridis.svg",
    "species": "tetraodon_nigroviridis",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0000948",
      "UBERON_0002113",
      "UBERON_0014892"
    ]
  },
  {
    "filename": "triticum_aestivum.flower_parts.svg",
    "species": "triticum_aestivum",
    "view": "flower_parts",
    "ids": [
      "PO_0020033",
      "PO_0020031",
      "PO_0009089",
      "PO_0009085",
      "PO_0009086",
      "PO_0009084",
      "PO_0009009",
      "PO_0009001",
      "PO_0009088",
      "PO_0009030",
      "PO_0009066"
    ]
  },
  {
    "filename": "triticum_aestivum.whole_plant.svg",
    "species": "triticum_aestivum",
    "view": "whole_plant",
    "ids": [
      "PO_0009047",
      "PO_0009005",
      "PO_0009049",
      "PO_0025034"
    ]
  },
  {
    "filename": "xenopus_tropicalis.svg",
    "species": "xenopus_tropicalis",
    "view": "",
    "ids": [
      "UBERON_0000955",
      "UBERON_0002113",
      "UBERON_0014892",
      "UBERON_0000948",
      "UBERON_0002107"
    ]
  },
  {
    "filename": "zea_mays.flower_parts.svg",
    "species": "zea_mays",
    "view": "flower_parts",
    "ids": [
      "PO_0009001",
      "PO_0020033",
      "PO_0020031",
      "PO_0020136",
      "PO_0025597",
      "PO_0009066",
      "PO_0009009",
      "PO_0009089",
      "PO_0009084"
    ]
  },
  {
    "filename": "zea_mays.whole_plant.svg",
    "species": "zea_mays",
    "view": "whole_plant",
    "ids": [
      "PO_0020126",
      "PO_0020136",
      "PO_0020142",
      "PO_0020127",
      "PO_0009005",
      "PO_0009074",
      "PO_0025034"
    ]
  }
]