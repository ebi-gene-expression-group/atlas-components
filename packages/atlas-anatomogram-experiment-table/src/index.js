export { default } from "./anatomogramExperimentTable.js";