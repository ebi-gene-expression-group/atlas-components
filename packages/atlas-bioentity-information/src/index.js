import BioentityInformation from './BioentityInformation.js'

export default BioentityInformation
