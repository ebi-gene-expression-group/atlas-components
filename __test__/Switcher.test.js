import React from 'react'
import renderer from 'react-test-renderer'
import { mount } from 'enzyme'

import Switcher from '../src/Switcher'

import svgsMetadata from '../src/json/svgsMetadata.json'
import {unique, getAnatomogramViews, getDefaultView, getParentView} from '../src/Assets'

const allSpecies = svgsMetadata
  .map((svgMetadata) => svgMetadata.species)
  .filter(unique)
  .sort()

const allChildrenViews = species =>
  svgsMetadata.filter(svgsMetadata => svgsMetadata.species === species)
    .filter(view => view.hasOwnProperty(`parent`))
    .map(data => data.view)

const organs = [`kidney`, `pancreas`, `liver`, `lung`, `gut`, `placenta`]

describe(`Anatomogram switcher`, () => {
  const requiredProps = {
    onChangeView: () => {}
  }

  test(`should contain as many buttons as views are defined for a species apart from organs`, function() {
    allSpecies.filter(species => !organs.includes(species)).forEach((species) => {
      expect(mount(<Switcher {...requiredProps} species={species}/>).find(`img`)).toHaveLength(getAnatomogramViews(species).length)
    })
  })
  
  test(`should contain a backward button for parent tissue parts for kidney species`, function() {
    const species = `kidney`
    allChildrenViews(species).forEach(view =>{
      expect(mount(<Switcher {...requiredProps} species={species} parentView={getParentView(species, view)}/>).find(`img`)).toHaveLength(1)
    }
    )
  })

  test(`should respond to onClick events with the anatomogram view and corresponding species`, () => {
    const onButtonClick = jest.fn()
    const wrapper = mount(<Switcher {...requiredProps} species={`homo_sapiens`} onChangeView={onButtonClick}/>)
    wrapper.find(`img`).first().simulate(`click`)
    wrapper.find(`img`).last().simulate(`click`)
    expect(onButtonClick.mock.calls.length).toBe(2)
    expect(onButtonClick.mock.calls[0]).toEqual([`homo_sapiens`, getAnatomogramViews(`homo_sapiens`)[0]])
    expect(onButtonClick.mock.calls[1]).toEqual([`homo_sapiens`, getAnatomogramViews(`homo_sapiens`)[2]])
  })

  allSpecies.forEach((species) => {
    test(`matches snapshot for ${species}`, () => {
      const tree = renderer.create(<Switcher {...requiredProps} species={species} selectedView={getDefaultView(species)}/>).toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
