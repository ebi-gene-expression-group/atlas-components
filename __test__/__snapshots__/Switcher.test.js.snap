// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Anatomogram switcher matches snapshot for anolis_carolinensis 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for arabidopsis_thaliana 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for bos_taurus 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for brachypodium_distachyon 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for gallus_gallus 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for gut 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for homo_sapiens 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for hordeum_vulgare 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for kidney 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for liver 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for lung 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for macaca_mulatta 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for monodelphis_domestica 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for mus_musculus 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for oryza_sativa 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for pancreas 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for papio_anubis 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for placenta 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for rattus_norvegicus 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for solanum_lycopersicum 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for solanum_tuberosum 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for sorghum_bicolor 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for tetraodon_nigroviridis 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for triticum_aestivum 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;

exports[`Anatomogram switcher matches snapshot for xenopus_tropicalis 1`] = `
<div
  className="sc-bdnylx eeRjTy"
/>
`;

exports[`Anatomogram switcher matches snapshot for zea_mays 1`] = `
<div
  className="sc-bdnylx eeRjTy"
>
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
  <img
    className="sc-gtssRu UZVzy"
    onClick={[Function]}
  />
</div>
`;
